api/v1/api.go

```diff
@@ -0,0 +1,83 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+)
+
+func path(endpoint string) string {
+	return fmt.Sprintf("/api/v1%s", endpoint)
+}
+
+func NewRESTApiV1(productionMode bool, logger *zap.Logger) *RESTApiV1 {
```

Comment:
[optional] Add error handling in case json is malformatted.

Reply:
Where exactly do you mean to add it? like in the `path` function or the `NewRESTApiV1` func?

Reply:
The APIs that receive JSON input, check it and if something is not right, they throw an error. For example check this one out: https://github.com/celestiaorg/bittwister/blob/main/api/v1/bandwidth.go#L14

Reply:
Interesting. I just received a 500 Internal Server error with no output.
But that might be due to wrong piping on my end.

Reply:
probably